crates/forge_analyzer/src/definitions.rs

```diff
@@ -226,9 +233,96 @@ pub fn run_resolver(
         module.visit_with(&mut collector);
     }
 
+    // This loop iterates through env's bodies and recreates an updated set of bodies to satisfy SSA form of IR dump.
+    let mut vars_map: HashMap<VarId, (VarKind, Option<VarId>, bool)> = HashMap::new();
+    for (func_id, body) in environment.bodies_mut().enumerate() {
+        // Builds the variables map to track existing all variables, store their VarId, and check if they have been assigned once (by SSA).
+        for var_id in body.vars.keys() {
+            let var_kind = *body.vars.get(var_id).unwrap();
+            vars_map.insert(var_id, (var_kind, None, false));
+        }
+        // Updates the instruction set with creation of new global reference variables if necessary.
+        for (block_id, block) in body.blocks.iter_mut_enumerated() {
+            for inst in block.iter_insts_mut() {
+                match inst {
+                    Inst::Assign(variable, rvalue) => {
+                        update_rvalue(rvalue, &vars_map);
+                        let var_id = variable.as_var_id().unwrap();
+                        let (var_kind, updated_var_id, global_exists) =
+                            vars_map.get_mut(&var_id).unwrap();
+                        if !*global_exists {
+                            *updated_var_id = Some(var_id);
+                            *global_exists = true;
+                        } else {
+                            let new_var_id = body.vars.push_and_get_key(*var_kind);
+                            variable.base = Base::Var(new_var_id);
+                            *updated_var_id = Some(new_var_id);
+                            *global_exists = true;
+                        }
+                    }
+                    Inst::Expr(rvalue) => {}
+                }
+            }
+        }
+        vars_map.clear();
+    }
     environment
 }
 
+// This function is a helper function to run_resolver() 's SSA Form Loop.
+// Input: rvalue and hashmap of VarIds that have been updated in the SSA Form Loop.
+// Output: new rvalue that has the newest VarId in its operands.
+pub fn update_rvalue(
+    rvalue: &mut Rvalue,
+    vars_map: &HashMap<VarId, (VarKind, Option<VarId>, bool)>,
+) {
+    match rvalue {
+        Rvalue::Unary(unop, Operand::Var(variable)) => {
+            let op_var_id = variable.as_var_id().unwrap();
+            if (vars_map.contains_key(&op_var_id)) && (vars_map.get(&op_var_id).unwrap().2) {
```

Comment:
```suggestion
            if let Some((_, updated_var_id, true)) = vars_map.get(&op_var_id) {
```

You can avoid the `contains_key` call by directly pattern matching, since [HashMap::get](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get) already does the check and returns `None` if the key isn't present. By pattern matching, you can also combine the check for checking if the global is present.